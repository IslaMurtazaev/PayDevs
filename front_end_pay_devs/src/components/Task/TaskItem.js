import React, { Component } from "react";
import { Link } from "react-router-dom";


class TaskItem extends Component {
  onClickDelete(taskId) {
    this.props.onDelete(taskId);
  }

  render() {
    const { task, projectId } = this.props;

    return (
      <div className="tasks">
        <h4 className="taskTitle"><b>Title:</b> {task.title}</h4>
        <h4 className="taskDescription"><b>Description:</b> {task.description}</h4>
        <h4 className="taskPrice"><b>Price:</b> {task.price}</h4>
        <h4 className="taskPaid">{task.paid ? "" : "not"} <b>paid</b></h4>
        <h4 className="taskCompleted"><b>{task.completed ? "Completed" : "Uncompleted"}</b></h4>
        <Link className="reactLink" to={`/project/${projectId}/Taskly/${task.id}/update`}>
          <button className="updateButton btn btn-warning" type="button">
            Update
          </button>
        </Link>
        <button type="button" className="btn btn-danger"
         onClick={this.onClickDelete.bind(this, task.id)}>Delete</button>
        </div>
    );
  }
}

export default TaskItem;
