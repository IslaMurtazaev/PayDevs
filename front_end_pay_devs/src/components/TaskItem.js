import React, { Component } from "react";
import {Link} from "react-router-dom";

class TaskItem extends Component {

    onClickDelete(taskId){
        this.props.onDelete(taskId);

    }

    render() {
        const { task, projectId } = this.props;

        return (
            <div>
                <h4>{task.title}</h4>
                <div>Description: {task.description}</div>
                <div>Price: {task.price}</div>
                <button><Link to={`/project/${projectId}/Taskly/${task.id}/update`}>UPDATE</Link></button>
                <button onClick={this.onClickDelete.bind(this, task.id)}>DELETE</button>
            </div>
        );
    }
}

export default TaskItem;