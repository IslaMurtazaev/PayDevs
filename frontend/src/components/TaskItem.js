import React, { Component } from "react";
import { Link } from "react-router-dom";


class TaskItem extends Component {
  onClickDelete(taskId) {
    this.props.onDelete(taskId);
  }

  render() {
    const { task, projectId } = this.props;

    return (
      <div className="task"> 
        <h4 className="taskTitle"><b>Title:</b> {task.title}</h4>
        <h5 className="taskDescription"><b>Description:</b> {task.description}</h5>
        <h5 className="taskPrice"><b>Price:</b> {task.price}</h5>
        <h5 className="taskPaid">{task.paid ? "" : "not"} <b>paid</b></h5>
        <h5 className="taskCompleted"><b>{task.completed ? "Completed" : "Uncompleted"}</b></h5>
        <Link className="reactLink" to={`/project/${projectId}/Taskly/${task.id}/update`}>
          <button type="button" className="updateButton btn btn-warning">
            Update
          </button>
        </Link>
        <button type="button" className="removeButton btn btn-danger"
         onClick={this.onClickDelete.bind(this, task.id)}>Delete</button>
        </div>
    );
  }
}

export default TaskItem;
