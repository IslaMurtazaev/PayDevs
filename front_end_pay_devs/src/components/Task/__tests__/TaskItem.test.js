import React from "react";
import TaskItem from "../TaskItem";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("<TaskItem />", () => {
  it("renders 1 <TaskItem />", () => {
    const task = { title: "Task number 1", description: "Test a ", price: 500 };
    const component = shallow(<TaskItem task={task} />);
    expect(component).toHaveLength(1);
  });

  it("renders attribs <TaskItem />", () => {
    const task = { title: "Task number 1", description: "Test a ", price: 500 };
    const component = shallow(<TaskItem task={task} />);
    expect(component.instance().props.task.title).toBe("Task number 1");
    expect(component.instance().props.task.description).toBe("Test a ");
    expect(component.instance().props.task.price).toBe(500);
  });

  it("renders mount <TaskItem /> h4 count", () => {
    const task = {
      id: 1,
      title: "Task number 1",
      description: "Test a ",
      paid: false,
      price: 500,
      completed: true
    };
    const component = shallow(<TaskItem task={task} />);
    expect(component.find("h4")).toHaveLength(1);
    expect(component.find("h5")).toHaveLength(4);
    expect(component.find(".taskTitle").text()).toBe(`Title: ${task.title}`);
    expect(component.find(".taskDescription").text()).toBe(
      `Description: ${task.description}`
    );
    expect(component.find(".taskPaid").text()).toBe("not paid");
    expect(component.find(".taskPrice").text()).toBe(`Price: ${task.price}`);
    expect(component.find(".taskCompleted").text()).toBe(
      task.completed ? "Completed" : "Uncompleted"
    );
  });

  it("renders mount <TaskItem /> click onDelete", () => {
    const onDeleteSpy = jest.fn();
    const task = {
      id: 1,
      title: "Task number 1",
      description: "Test a ",
      price: 500
    };
    const component = shallow(<TaskItem task={task} onDelete={onDeleteSpy} />);
    component.find(".btn-danger").simulate("click");
    expect(onDeleteSpy).toBeCalledWith(1);
  });

  it("matches previous snapshot", () => {
    const task = {
      id: 1,
      title: "Task number 1",
      description: "Test a ",
      price: 500
    };
    expect(toJson(shallow(<TaskItem task={task} />))).toMatchSnapshot();
  });
});
