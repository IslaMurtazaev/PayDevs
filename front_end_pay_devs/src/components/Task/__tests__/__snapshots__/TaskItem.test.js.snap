// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskItem /> matches previous snapshot 1`] = `
<div
  className="tasks"
>
  <h4
    className="taskTitle"
  >
    <b>
      Title:
    </b>
     
    Task number 1
  </h4>
  <h4
    className="taskDescription"
  >
    <b>
      Description:
    </b>
     
    Test a 
  </h4>
  <h4
    className="taskPrice"
  >
    <b>
      Price:
    </b>
     
    500
  </h4>
  <h4
    className="taskPaid"
  >
    not
     
    <b>
      paid
    </b>
  </h4>
  <h4
    className="taskCompleted"
  >
    <b>
      Uncompleted
    </b>
  </h4>
  <Link
    className="reactLink"
    replace={false}
    to="/project/undefined/Taskly/1/update"
  >
    <button
      className="updateButton btn btn-warning"
      type="button"
    >
      Update
    </button>
  </Link>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    type="button"
  >
    Delete
  </button>
</div>
`;
